pkg/util/policy/agent/agent.py

```diff
@@ -284,138 +258,231 @@ def make_new_full_ruleset(current_rules, new_rules):
-    rules[tenant_policy_vector_chain] = [
-        _make_rule(tenant_policy_vector_chain, "-j %s" % target_segment_forward_chain),
-        _make_rule(tenant_policy_vector_chain, "-j %s" % from_segment_forward_chain),
-        _make_rule(tenant_policy_vector_chain, "-j DROP")
-    ]
-
-    # Jump from per-segment chain into policy chain
-    rules[target_segment_forward_chain] = [
-        _make_rule(target_segment_forward_chain, "-j %s" % policy_chain_name),
-        _make_rule(target_segment_forward_chain, '-m comment --comment POLICY_CHAIN_HEADER -j RETURN')
-    ]
-    rules[from_segment_forward_chain] = [
-        _make_rule(from_segment_forward_chain, "-j %s" % policy_chain_name),
-        _make_rule(from_segment_forward_chain, '-m comment --comment POLICY_CHAIN_HEADER -j RETURN')
-    ]
-
-    # Assemble the rules for the top-level policy chain. These rules look at
-    # the IP addresses (source and dest) and figure out whether this is
-    # incoming our outgoing traffic.
-    u32_in_match  = _make_u32_match(addr_scheme, tenant, from_segment,
-                                    tenant, target_segment)
-    u32_out_match = _make_u32_match(addr_scheme, tenant, target_segment,
-                                    tenant, from_segment)
-
-    in_chain_name  = policy_chain_name[:-1] + "-IN_"
-    out_chain_name = policy_chain_name[:-1] + "-OUT_"
-
-    rules[policy_chain_name] = [
-        _make_rule(policy_chain_name, '-m u32 --u32 "%s" -j %s' %
-                                           (u32_in_match, in_chain_name)),
-        _make_rule(policy_chain_name, '-m u32 --u32 "%s" -j %s' %
-                                           (u32_out_match, out_chain_name)),
-        _make_rule(policy_chain_name, '-m comment --comment PolicyId=%s -j RETURN' % policy_id)
-    ]
-
-    # Assemble the rules for the port/protocol portion of the policy.
-    # Note! We currently only support TCP with a given port. No ICMP and no
-    # UDP!
-
-    rules[in_chain_name] = [
-        _make_rule(in_chain_name,
-                   '-p tcp --dport %d --tcp-flags SYN SYN -j ACCEPT' %
-                   policy_def['allowFrom']['port']),
-        _make_rule(in_chain_name,
-                   '-m state --state ESTABLISHED -j ACCEPT'),
-        _make_rule(in_chain_name, '-j RETURN')
-    ]
-
-    rules[out_chain_name] = [
-        _make_rule(out_chain_name,
-                   '-p tcp --sport %d --tcp-flags SYN,ACK SYN,ACK -j ACCEPT' %
-                   policy_def['allowFrom']['port']),
-        _make_rule(out_chain_name,
-                   '-m state --state ESTABLISHED -j ACCEPT'),
-        _make_rule(out_chain_name, '-j RETURN')
-    ]
+
+    # We really need a list of chains but using dict here
+    # to avoid extra checks for item in list.
+    policy_chains = {}
+
+    # Create chain names for each target and stuff them with default rules
+    name = policy_def['name']
+    for target in policy_def.get('applied_to'):
```

Comment:
This code is called after validation has taken place, right? So we can write:

    policy_def["applied_to"]

instead of

    policy_def.get("applied_to")

Same for the other fields below.